Replace pip-tools with uv

meshy opened the pull request:
Before now, we installed development requirements with `pip`, and compiled them from the `pyproject.toml` with `pip-tools`.

Recently, `uv` was released, which boasts a significant speed advantage over those tools.

There are a couple of minor consequences:

- We're obliged to set a minimum version of the installed packages.
  This is of little concern because we're already pinning the dev package versions.

- It appears that the installation of the `tomli` dev requirement is no longer conditional.
  This is also of little concern, given how much faster everything installs now.

- We have to explicitly create a virtual env in CI before installing.
  This is probably the right thing to do anyway.

<table>
<tr>
 <td>Before:
 
![image](https://github.com/kraken-tech/django-integrity/assets/767671/7aa1d630-9e20-4dd0-a124-22b18524965c)

  </td>
 <td>After:
 
![image](https://github.com/kraken-tech/django-integrity/assets/767671/ab09c22c-5aab-407f-97ef-8bcc4e0b7265)


   </td>
</table>

Fixes #19.

delfick commented on makefile:
```diff
@@ -16,6 +17,10 @@ help:
 
+.PHONY:install-ci
+install-ci: install_prerequisites
+	uv pip install --system --requirement requirements/development.txt
```
why not use a virtualenv for ci too?

meshy replied:
I rejected this idea to start with, but now that I've slept on it, I think it makes mroe sense to create the virtualenv in CI. I'll switch over to that.

delfick reviewed (dismissed):
nice

jarshwah commented on makefile:
```diff
@@ -27,11 +31,7 @@ lint: format style typing
-.PHONY:package
-package:
-	pip wheel .
+	uv pip compile pyproject.toml --quiet --upgrade --resolver=backtracking --extra=dev --output-file=requirements/development.txt --unsafe-package django
```
Curious, why haven't we got django linked as a dependency in our prproject deps?

meshy replied:
I developed an aversion to adding Django to requirements long ago, but I must admit that I can't quite remember why now.

In this particular case I made sure to exclude it because it makes testing against different versions of Django easier with Tox.

Thankfully there's no reason to install this project without Django, so I can't imagine it causing an issue.

meshy replied:
This situation is fixed and clarified in https://github.com/kraken-tech/django-integrity/pull/30

codeinthehole commented on CHANGELOG.md:
```diff
@@ -4,6 +4,8 @@ All notable changes to this project will be documented in this file.
 
 ## Unreleased changes
```
Should the v0.1 section be filed under "Unreleased changes"? It's on PyPI right?

meshy replied:
It's a sibling, rather than a subheading. I think this is OK because this is a living document and new changes are likely to be added to this section a lot.

This file doesn't get sent to PyPI directly -- instead there's a link to the file in this repo.

meshy commented:
This work has revealed some shortcomings in how requirements were handled.

I've refactored them in https://github.com/kraken-tech/django-integrity/pull/30 and will rebase this PR once that's in.

meshy commented:
I've addressed the aforementioned issues, and updated.

You may notice that the speed improvements in the screenshots are now more modest -- this is mainly the result of setting up caching in #30.

I'm asking you for re-review because the commits have changed a fair bit since they were last approved.

delfick reviewed (approved):
if you do `python -m pip install uv==0.1.40 -t here --disable-pip-version-check` you can then do `./here/bin/uv` to get a uv for making the venv with and then use it with the `-p` option to the python in the venv to install requirements into that venv.

meshy commented:
> if you do `python -m pip install uv==0.1.40 -t here --disable-pip-version-check` you can then do `./here/bin/uv` to get a uv for making the venv with and then use it with the `-p` option to the python in the venv to install requirements into that venv.

I think I'm happy with this approach for now, but that's good to know, thanks!
